lib/scsi_defs.h

```diff
@@ -108,5 +108,68 @@ namespace scsi_defs {
   } ABSL_ATTRIBUTE_PACKED;
   static_assert(sizeof(ReadCapacity10Data) == 8);
 
+// SCSI Reference Manual Table 76 https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  struct PersistentReserveInCommand {
+    OpCode op_code = OpCode::kPersistentReserveIn;
+    uint8_t reserved_1 : 3;
+    uint8_t service_action : 5;
+    uint64_t reserved_2 : 40;
+    uint16_t allocation_length;
+    ControlByte control_byte;
+  } ABSL_ATTRIBUTE_PACKED;
+  static_assert(sizeof(PersistentReserveInCommand) == 10);
+
+  // Persistent Reserve In Read Reservation Data No Reservation
+  // SCSI Reference Manual Table 79 https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  struct PriReadReservationDataNoReservation {
+    uint32_t prgeneration;
+    uint32_t additional_length;
+  } ABSL_ATTRIBUTE_PACKED;
+  static_assert(sizeof(PriReadReservationDataNoReservation) == 8);
+
+  // Persistent Reserve In Read Reservation Data With Reservation
+  // SCSI Reference Manual Table 80 https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  struct PriReadReservationDataWithReservation {
+    PriReadReservationDataNoReservation priDataNoReservation;
+    uint64_t reservation_key;
+    uint32_t obsolete_1;
+    uint8_t reserved;
+    uint8_t scope : 4;
+    uint8_t type : 4;
+    uint16_t obsolete_2;
+  } ABSL_ATTRIBUTE_PACKED;
+  static_assert(sizeof(PriReadReservationDataWithReservation) == 24);
+
+  // SCSI Reference Manual Table 88 https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  struct PersistentReserveOutCommand {
+    OpCode op_code = OpCode::kPersistentReserveOut;
+    uint8_t reserved_1 : 3;
+    uint8_t service_action : 5;
+    uint8_t scope : 4;
+    uint8_t type : 4;
+    uint16_t reserved_2;
+    uint32_t parameter_list_length;
+    ControlByte control_byte;
+  } ABSL_ATTRIBUTE_PACKED;
+  static_assert(sizeof(PersistentReserveOutCommand) == 10);
+
+  // Persistent Reserve Out Parameter List
+  // Used by Persistent Reserve Out command for any service action other than Register and Move
+  // SCSI Reference Manual Table 90 https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  struct ProParamList {
+    uint64_t reservation_key;
+    uint64_t service_action_reservation_key;
+    uint32_t obsolete_1;
```

Comment:
nit: In some of our previous PRs, we are mentioning the number of bits used by the field even if it is using all bits of the aforementioned type. I suggest having a standard for this.

Reply:
I can see how specifying the bits for all fields can introduce clarity but I wonder if it might unnecessarily clutter the code. Since we assert the size of each struct, I also don't think this will add much safety. I'd love to hear more views!

Reply:
I think Emily is right that doing so doesn't really add meaningful size safety. So, the choice is mostly stylistic. I could go either way.

Reply:
Yeah, I agree that there's no upside to it in terms of logic. I am fine going either way too but we should fix this before merging our branches so that our code is consistent everywhere

Reply:
i updated my changes so they specify the size!

Reply:
Just for future reference: https://stackoverflow.com/questions/25822679/packed-bit-fields-in-c-structures-gcc